fs/layer/node.go

```diff
@@ -112,6 +113,47 @@ var fuseOpFailureMetrics = map[string]string{
 	fuseOpWhiteoutGetattr: commonmetrics.FuseWhiteoutGetattrFailureCount,
 }
 
+type FuseOperationCounter interface {
+	Inc(string)
+	Run()
+}
+
+type defaultFuseOperationCounter struct {
+	opCounts    map[string]*atomic.Int32
+	waitPeriod  time.Duration
+	imageDigest digest.Digest
+}
+
+func NewDefaultOperationCounter(imgDigest digest.Digest, waitPeriod time.Duration) FuseOperationCounter {
+	f := &defaultFuseOperationCounter{
+		imageDigest: imgDigest,
+		waitPeriod:  waitPeriod,
+		opCounts:    make(map[string]*atomic.Int32),
+	}
+	f.opCounts[fuseOpGetattr] = new(atomic.Int32)
+	f.opCounts[fuseOpGetxattr] = new(atomic.Int32)
+	f.opCounts[fuseOpListxattr] = new(atomic.Int32)
+	f.opCounts[fuseOpLookup] = new(atomic.Int32)
+	f.opCounts[fuseOpOpen] = new(atomic.Int32)
+	f.opCounts[fuseOpReaddir] = new(atomic.Int32)
+	f.opCounts[fuseOpFileRead] = new(atomic.Int32)
+	f.opCounts[fuseOpFileGetattr] = new(atomic.Int32)
+	f.opCounts[fuseOpWhiteoutGetattr] = new(atomic.Int32)
```

Comment:
This feels like it should have a less repetitive representation. A loop through a slice of operation names would be a bit blunt, but I can't immediately think of a better option.